secrets/secrets.go

```diff
@@ -298,3 +320,32 @@ func NewSecrets(r io.Reader) (*Secrets, error) {
 	}
 	return secrets, nil
 }
+
+func csiPathParser(path string) (Document, error) {
+	secretsDocument := Document{
+		Secrets: make(map[string]GenericSecret),
+	}
+	err := filepath.Walk(
+		path,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.Mode().IsDir() {
+				return nil
+			}
+			// parse file
+			file, err := os.Open(path)
+			if err != nil {
+				return err
+			}
+			var secretFile CSIFile
+			err = json.NewDecoder(file).Decode(&secretFile)
+			if err != nil {
+				return err
+			}
+			secretsDocument.Secrets[path] = secretFile.Secret
```

Comment:
🔕 I wonder if we want to guard against the case that the same `path` appearing in different files. e.g. check that `path` is not already in the map and return an error if that's the case:

```go
if _, ok := secretsDocument.Secrets[path]; ok {
  return someError
}
secretsDocument.Secrets[path] = secretFile.Secret
```

Reply:
The Vault CSI provider should ensure that two files refer to different secrets.  It creates one file per secret path. So in this example the secrets
`secret/creds/user` and `secret/creds/pass`
should appear in a directory like
```
secret
└── creds
    ├── user
    └── pass
```

Duplicate paths shouldnt show up